pkg/internal/loader/disk/disk_test.go

```diff
@@ -195,13 +195,136 @@ metadata:
 				if test.namespace != nil {
 					t.Setenv("NAMESPACE", *test.namespace)
 				}
-
-				loader := New[compapi.Component](tmp)
+				loader := NewComponents(tmp)
 				components, err := loader.Load(context.Background())
-
 				assert.Equal(t, test.expErr, err != nil, "%v", err)
 				assert.Equal(t, test.expComps, components)
 			})
 		}
 	})
 }
+
+func Test_loadWithOrder(t *testing.T) {
+	t.Run("no file should return empty set", func(t *testing.T) {
+		tmp := t.TempDir()
+		d := NewComponents(tmp).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Empty(t, set.order)
+		assert.Empty(t, set.ts)
+	})
+
+	t.Run("single manifest file should return", func(t *testing.T) {
+		tmp := t.TempDir()
+		require.NoError(t, os.WriteFile(filepath.Join(tmp, "test-component.yaml"), []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+`), fs.FileMode(0o600)))
+
+		d := NewComponents(tmp).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Equal(t, []manifestOrder{
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 0},
+		}, set.order)
+		assert.Len(t, set.ts, 1)
+	})
+
+	t.Run("3 manifest file should have order set", func(t *testing.T) {
+		tmp := t.TempDir()
+		require.NoError(t, os.WriteFile(filepath.Join(tmp, "test-component.yaml"), []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
```

Comment:
Can these names be different to avoid confusion since this isn't valid config in Dapr? It confuses the example you are testing since the `metadata.name` && `spec.type` are the same for each but the `manifestIndex` is different each time. Or make a note that its not valid config but its testing the indexing properly?